### ISSUE TITLE
index.md based folder note meta data not recognized

### ISSUE THREAD

**Issue by octo-kumo** [SebastianMC/obsidian-custom-sort#156]
Hi I have the following config in `/index.md`.
```markdown
---
sorting-spec: |
  target-folder: ctf
  < a-z by-metadata: created
---
```

I want to sort the folders under `/ctf`, they all have `index.md` so the final path is something like `/ctf/2024-ictf/index.md`.
Those files have frontmatter `created` but the sorting isn't working for some reason.

I know that the sorting is taking some effect because my other files under `/ctf` are changing order, just not the folders.

**Comment by SebastianMC** [comment]
Hi @octo-kumo 

some questions arise when I read your question. Instead of jumping directly into a series of clarifying questions & answers interchange, let me guess your setup and what you want to achieve.

## Guess 1:
Assume you want to keep a single sorting spec configuration in a single place in a single file and let the sorting specification be applied to all subfolders of `/ctf`.

In that case you should employ the following specification (and keep it in a single place):

```yaml
---
sorting-spec: |
  target-folder: ctf/*
  < a-z by-metadata: created
---
```

The above spec will be applied to the `ctf` folder and all of its descendant subfolders (`  ctf/*` wildard syntax is used for that).

## Guess 2:
Assume you want to keep a separate sorting spec in each of `index.md` files in the subfolders for which you want to apply the sorting.

In that case each of the relevant `index.md` should include in frontmatter the spec like:

```yaml
---
sorting-spec: |
  < a-z by-metadata: created
---
```

When you omit the `  target-folder: .........`  specification the sorting is applied to the folder in which the sorting specification resides.

## ---

As you can see the specs for both guesses are variants of your original spec with minor addition / removal. That means you were very close!!!

If my guesses were not exactly what you'd expect, please come back with more details.

**Comment by octo-kumo** [comment]
After each change I did the right click > apply custom sorting.

Neither of the settings worked.

![image](https://github.com/user-attachments/assets/49756978-3c5a-422a-9b17-8df965c6ef51)
![image](https://github.com/user-attachments/assets/732fec15-741c-4bae-b8c6-dde67b37f3a8)

**Comment by SebastianMC** [comment]
Hi @octo-kumo 

thanks for providing more details and the screenshots, now the scenario is clear!

Knowing exactly what you want to achieve it looks there is a bug in this edge scenario.
For now, only the folder notes named exactly as the parent folder variant of _folder notes_ is supported when scanning for folder's metadata. The _index file folder notes_ aren't taken into account.

So, even if you correctly configure the plugin for _index based folder notes_ (which you must have done):

<img width="759" alt="2024-08-18_00-05-58" src="https://github.com/user-attachments/assets/6659f8ac-7218-4e8b-8662-4aa60fa300c1">

this configuration only tells the plugin to read sorting configuration from the index files, not the folder's metadata.

The fix to this bug is not trivial, when looking from the perspective of backward compatibility.

The configuration box in custom sort settings window allows entering multiple names and/or paths of notes to scan for sorting specifications. Thus, using that field to specify the index file is only one of its supported functions. In other words, if you enter some note names / paths, the plugin doesn't know if these are related to _folder notes_ or not.

I need to find out some backward-compatible not-confusing smart way of extending the settings window, so that you can tell the plugin that _index-file-based_ approach to _folder notes_ is employed in the vault. Plus confirmation of the index file name (if different than _index.md_). So, a new checkbox and a new text box plus smart logic not to break backward-compatibility.

What do you think?

**Comment by octo-kumo** [comment]
Sounds good

**Comment by SebastianMC** [comment]
The improvement is available in the recent [release 2.1.13](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/2.1.13) 

@octo-kumo please retest and let me know if it works for you

**Comment by octo-kumo** [comment]
yeb it works now thanks!
